varys/producer.py

```diff
@@ -60,12 +60,12 @@ def _on_connection_closed(self, _unused_connection, reason):
             self._log.warning(
                 f"Connection to broker closed, will attempt to re-connect in 10 seconds: {reason}"
             )
-            self._connection.ioloop.call_later(10, self._connection.ioloop.start)
+            self._connection.ioloop.call_later(10, self._connection.ioloop.stop)
 
     def _on_channel_closed(self, channel, reason):
         self._log.warning(f"Channel {channel} was closed by broker: {reason}")
         self._channel = None
-        if not self._stopping:
+        if not self._stopping and not self._connection.is_closed:
```

Comment:
Not in the examples but at least avoids us trying to close a connection that is already closed. (I don't think the examples are detailed enough to consider the server restarting.)